tgui/packages/tgui/interfaces/TeleConsole/index.tsx

```diff
@@ -28,41 +35,139 @@ export const TeleConsole = (_props, context) => {
-          onDeleteBookmark={handleDeleteBookmark}
-          onRestoreBookmark={handleRestoreBookmark}
-          targetCoords={[xTarget, yTarget, zTarget]}
-        />
-        <ConnectionSection
-          isConnected={isConnectedToHost}
-          isPanelOpen={!!isPanelOpen}
-          onCyclePad={handleCyclePad}
-          onReset={handleResetConnect}
-          onRetry={handleRetryConnect}
-          padNum={padNum}
-        />
+        <Stack vertical fill>
+          <Stack.Item>
+            <Tabs>
+              <Tabs.Tab
+                icon="box"
+                selected={tab === Tab.Local}
+                onClick={() => setTab(Tab.Local)}
+              >
+                Local
+              </Tabs.Tab>
+              <Tabs.Tab
+                icon="wrench"
+                selected={tab === Tab.LongRange}
+                onClick={() => setTab(Tab.LongRange)}
+              >
+                Long Range
+              </Tabs.Tab>
+            </Tabs>
+          </Stack.Item>
+
+          {(tab===Tab.Local) && (
+            <Stack.Item>
+              <Section>
+                <CoordinatesSection />
+                <Section>
+                  <Button icon="sign-out-alt" onClick={() => act('send')} disabled={!isConnectedToHost}>
+                    Send
+                  </Button>
+                  <Button icon="sign-in-alt" onClick={() => act('receive')} disabled={!isConnectedToHost}>
+                    Receive
+                  </Button>
+                  <Button onClick={() => act('portal')} disabled={!isConnectedToHost}>
+                    <Icon name="ring" rotation={90} />
+                    Toggle Portal
+                  </Button>
+                  <Button icon="magnifying-glass" onClick={() => act('scan')} disabled={!isConnectedToHost}>
+                    Scan
+                  </Button>
+                </Section>
+                {readout && <Section>{formatReadout(readout)}</Section>}
+                <BookmarksSection
+                  bookmarks={bookmarks}
+                  maxBookmarks={maxBookmarks}
+                  onAddBookmark={handleAddBookmark}
+                  onDeleteBookmark={handleDeleteBookmark}
+                  onRestoreBookmark={handleRestoreBookmark}
+                  targetCoords={[xTarget, yTarget, zTarget]}
+                />
+                <ConnectionSection
+                  isConnected={isConnectedToHost}
+                  isPanelOpen={!!isPanelOpen}
+                  onCyclePad={handleCyclePad}
+                  onReset={handleResetConnect}
+                  onRetry={handleRetryConnect}
+                  padNum={padNum}
+                />
+              </Section>
+            </Stack.Item>
+          )}
+          {(tab===Tab.LongRange) && (
+            <Stack.Item>
+              <Section>
+                <Section title="Destinations">
+                  <LabeledList>
+                    {destinations.length ? destinations.map((d) => (
```

Comment:
Looks like you only use the `name` here, consider doing as:
```suggestion
                    {destinations.length ? destinations.map(({ name }) => (
```
Then you can just refer to `name` instead of `d.name` (or `d["name"]`), and it also lets you write object instantiation as a shorthand (`act('action', { name })`)